test/receiverSpec.js

```diff
@@ -288,6 +303,28 @@ describe('Receiver Module', function () {
           expect(results).to.deep.equal(['sent'])
         })
 
+        it('should ignore transfers without packet.data', function * () {
+          delete this.transfer.data.ilp_header.data
+          const request = this.receiver.createRequest({
+            amount: 1,
+            id: '22e315dc-3f99-4f89-9914-1987ceaa906d'
+          })
+          const results = yield this.client.emitAsync('incoming_prepare', this.transfer)
+          expect(results).to.deep.equal(['condition-mismatch'])
+        })
+
+        it('should fulfill conditions of transfers packet.data.expires_at but no packet.data.data', function * () {
+          const request = this.receiver.createRequest({
+            amount: 1,
+            id: '22e315dc-3f99-4f89-9914-1987ceaa906d',
+            expiresAt: this.transfer.data.ilp_header.data.expires_at
+          })
+          delete this.transfer.data.ilp_header.data.data
```

Comment:
If you want to test what happens when the packet is modified and the data is tampered with, you should include the data in the `createRequest` params and then delete it